publication-rest/src/main/java/no/unit/nva/publication/delete/DeletePublicationHandler.java

```diff
@@ -33,22 +37,45 @@ public DeletePublicationHandler() {
      * @param resourceService resourceService
      * @param environment     environment
      */
-    public DeletePublicationHandler(ResourceService resourceService, Environment environment) {
+    public DeletePublicationHandler(ResourceService resourceService,
+                                    Environment environment,
+                                    IdentityServiceClient identityServiceClient) {
         super(Void.class, environment);
         this.resourceService = resourceService;
+        this.identityServiceClient = identityServiceClient;
     }
-    
+
     @Override
     protected Void processInput(Void input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
         SortableIdentifier identifier = RequestUtil.getIdentifier(requestInfo);
-        String owner = requestInfo.getNvaUsername();
-        URI customerId = requestInfo.getCurrentCustomer();
-        UserInstance userInstance = UserInstance.create(owner, customerId);
+        var userInstance = createUserInstanceFromRequest(requestInfo);
         
         resourceService.markPublicationForDeletion(userInstance, identifier);
-        
         return null;
     }
+
+    private UserInstance createUserInstanceFromRequest(RequestInfo requestInfo) throws UnauthorizedException {
+        if (requestInfo.clientIsThirdParty()) {
+            return createUserInstanceForExternalClientUser(requestInfo);
+        } else {
+            var owner = requestInfo.getNvaUsername();
+            var customerId = requestInfo.getCurrentCustomer();
+            return UserInstance.create(owner, customerId);
+        }
+    }
+
+    private UserInstance createUserInstanceForExternalClientUser(RequestInfo requestInfo) throws UnauthorizedException {
```

Comment:
Is this duplicated from CreatePublicationHandler? If so, have you considered a shared util class or even nva-commons?